src/utilities/alfalfa/AlfalfaGlobalVariable.cpp

```diff
@@ -0,0 +1,42 @@
+#include "AlfalfaGlobalVariable.hpp"
+
+#include "utilities/idd/IddObject.hpp"
+#include "utilities/idd/IddEnums.hpp"
+
+#include <utilities/idd/EnergyManagementSystem_GlobalVariable_FieldEnums.hxx>
+#include <utilities/idd/OS_EnergyManagementSystem_GlobalVariable_FieldEnums.hxx>
+#include <utilities/idd/IddEnums.hxx>
+
+#include <fmt/format.h>
+
+namespace openstudio {
+namespace alfalfa {
+  AlfalfaGlobalVariable::AlfalfaGlobalVariable(const std::string& variable_name)
+    : AlfalfaComponent("GlobalVariable", Capability::Output | Capability::Input) {
+    if (variable_name.size() == 0) {
+      throw std::runtime_error("Error creating AlfalfaGlobalVariable: variable_name must be non-empty");
+    }
+    parameters["variable_name"] = variable_name;
+  }
+
+  AlfalfaGlobalVariable::AlfalfaGlobalVariable(const IdfObject& global_variable)
+    : AlfalfaComponent("GlobalVariable", Capability::Output | Capability::Input) {
+    IddObjectType idd_type = global_variable.iddObject().type();
+    boost::optional<std::string> variable_name;
+
+    if (idd_type == IddObjectType::OS_EnergyManagementSystem_GlobalVariable) {
+      variable_name = global_variable.getString(OS_EnergyManagementSystem_GlobalVariableFields::Name);
+    } else if (idd_type == IddObjectType::EnergyManagementSystem_GlobalVariable) {
+      variable_name = global_variable.getString(EnergyManagementSystem_GlobalVariableExtensibleFields::ErlVariableName);
+    } else {
```

Comment:
K so second time I see this pattern so it's a thing. We have one Ctor that takes IdfObject and tests if this is a ModelObject type or an actual IdfObject/Workspace object.

This is a couple of types, so it's probably fine... and since you're living inside utilities/ you don't have access to Model namespace really.

If you move up one folder as @kbenne asked about, then I'd probably specialize the ctor for `modelObject` and have it call the getters instead of getString

Reply:
I'm happy to move it up a namespace, I just didn't want to be too entitled. I actually started with it on the top level and then moved it here later.